gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageGrpcWriteChannel.java

```diff
@@ -373,22 +520,23 @@ private WriteObjectRequest getCachedRequest(
     the GCS backend. The increase will be linear to number of chunks written, so that
     should also be fine.    */
     private void freeUpCommittedRequests(
```

Comment:
Can we break it into two PRs:
1. To make freeUpCommittedRequests asynchronous and optimization around it.
2. Avoiding memory copy for writes.

Reply:
@abmodi I have removed the `freeUpCommittedRequests asynchronous` optimization from this PR. Once this is merged, someone can pick up the changes from the commit history of this PR and apply on top of that. Hope this is fine.